### ISSUE TITLE
jsx-no-target-blank fails when spread props is before target even when warnOnSpreadAttributes is false

### ISSUE THREAD

**Issue by DatDinoDoe** [yannickcr/eslint-plugin-react#2952]
Starting in 7.23.0, the jsx-no-target-blank rule will fail when `warnOnSpreadAttributes` is false and there's a spread before a non-`_blank` target, which is supposed to pass even if `warnOnSpreadAttributes` is true.

Example:
When `warnOnSpreadAttributes` is false (or default):
```javascript
 <a href='some-link' {...otherProps} target='some-non-blank-target'>
```
Would fail, while
```javascript
 <a href='some-link' target='some-non-blank-target' {...otherProps} >
```
would pass. If `warnOnSpreadAttributes` was true, both would fail.

I believe the issue is the `if/elseif` logic here:
https://github.com/yannickcr/eslint-plugin-react/blob/v7.23.1/lib/rules/jsx-no-target-blank.js#L149

**Comment by ljharb** [comment]
cc @Nokel81
